src/kraken/core/system/graph.py

```diff
@@ -158,8 +161,34 @@ def _remove_nodes_keep_transitive_edges(self, nodes: Iterable[str]) -> None:
             self._digraph.remove_node(task_path)
 
     def _get_ready_graph(self) -> DiGraph:
-        """Updates the ready graph."""
-        return restricted_view(self._digraph, self._completed_tasks, set())
+        """Updates the ready graph. Remove all completed tasks and any non-strict dependencies (edges) on failed
+        tasks."""
+
+        removable_edges: set[tuple[str, str]] = set()
+
+        def set_non_strict_edge_for_removal(u: str, v: str) -> None:
+            nonlocal removable_edges
+            out_edge = not_none(self._get_edge(u, v))
+            if not out_edge.strict:
+                removable_edges.add((u, v))
+
+        for failed_task_path in self._failed_tasks:
+            for out_task_path in self._digraph.successors(failed_task_path):
+                out_task_data = self._digraph.nodes[out_task_path]["data"]
```

Comment:
```suggestion
                out_task = self._digraph.nodes[out_task_path]["data"]
```

nit: Would suggest to just call this `out_task` since that resembles the type of the variable more concretely.